test/deployframework/helpers.go

```diff
@@ -0,0 +1,104 @@
+package deployframework
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	meteringv1 "github.com/operator-framework/operator-metering/pkg/apis/metering/v1"
+	v1 "k8s.io/api/core/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (df *DeployFramework) createResourceDirs(path string) ([]string, error) {
+	envVarArr := []string{
+		"METERING_TEST_NAMESPACE=" + df.Config.Namespace,
+	}
+
+	testDirsMap := map[string]string{
+		logDir:              "LOG_DIR",
+		reportsDir:          "REPORTS_DIR",
+		meteringconfigDir:   "METERINGCONFIGS_DIR",
+		datasourcesDir:      "DATASOURCES_DIR",
+		reportqueriesDir:    "REPORTQUERIES_DIR",
+		hivetablesDir:       "HIVETABLES_DIR",
+		prestotablesDir:     "PRESTOTABLES_DIR",
+		storagelocationsDir: "STORAGELOCATIONS_DIR",
+	}
+
+	for dirname, env := range testDirsMap {
+		dirPath := filepath.Join(path, dirname)
+
+		err := os.MkdirAll(dirPath, 0777)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create the directory %s: %v", dirPath, err)
+		}
+
+		envVarArr = append(envVarArr, env+"="+dirPath)
+	}
+
+	return envVarArr, nil
+}
+
+func (df *DeployFramework) logPollingSummary(targetPods int, readyPods []string, unreadyPods []podStat) {
+	df.Logger.Infof("Poll Summary")
+	df.Logger.Infof("Current ratio of ready to target pods: %d/%d", len(readyPods), targetPods)
+
+	for _, unreadyPod := range unreadyPods {
+		if unreadyPod.Total == 0 {
+			df.Logger.Infof("Pod %s is pending", unreadyPod.PodName)
+			continue
+		}
+		df.Logger.Infof("Pod %s has %d/%d ready containers", unreadyPod.PodName, unreadyPod.Ready, unreadyPod.Total)
+	}
+}
+
+func (df *DeployFramework) checkPodStatus(pod v1.Pod) (bool, int) {
+	if pod.Status.Phase != v1.PodRunning {
+		return false, 0
+	}
+
+	var unreadyContainers int
+
+	for _, status := range pod.Status.ContainerStatuses {
+		if !status.Ready {
+			unreadyContainers++
+		}
+	}
+
+	return unreadyContainers == 0, len(pod.Status.ContainerStatuses) - unreadyContainers
+}
+
+func (df *DeployFramework) addTestingNamespaceLabel(namespace, labelVal string) error {
```

Comment:
Ideally we could just set this when we create the namespace we're using for the test.

Reply:
The namespace is created using the deploy package so that would involve altering that package to support passing a label to the `deployer.installNamespace()` method, but that method is also private (so `deployer.Install()`).

Reply:
I see. I think it would be ideal if we could set extra labels by passing a map to the deployer.New constructor or deployer config.